src/useElementRect/useElementRect.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable id-length */
+import { ref, Ref, watchEffect } from 'vue'
+import { useResizeObserver, UseResizeObserverCallback } from '../useResizeObserver/useResizeObserver'
+
+type ElementRect = {
+  height: Ref<number>,
+  width: Ref<number>,
+  x: Ref<number>,
+  y: Ref<number>,
+  left: Ref<number>,
+  top: Ref<number>,
+  right: Ref<number>,
+  bottom: Ref<number>,
+}
+
+export function useElementRect(element: HTMLElement | undefined | Ref<HTMLElement | undefined>): ElementRect {
```

Comment:
nit: I don't think there's a reason to accept undefined if its not a ref.

Reply:
I just copied useElementWidth, should I change it there as well?

Reply:
yeah I think so? Unless you can think of a reason to accept undefined. We only accept Ref<undefined> as a convenience since template refs are undefined during setup